app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user, foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  def update_post_counter(user)
```

Comment:
[OPTIONAL]
- You could use [rails active callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) to call this method every time a new post is saved to the database. After that, you can make the method private. It can look like this
```
  after_save :update_post_counter

  private

  def update_post_counter(user)
    user.post_counter = Post.where(author_id: user.id).count
  end
```